Frameworks/include/CGIWICBitmap.h

```diff
@@ -161,34 +225,43 @@ class CGIWICBitmap
     }
 
     HRESULT STDMETHODCALLTYPE GetResolution(_Out_ double* dpiX, _Out_ double* dpiY) {
-        UNIMPLEMENTED();
-        return E_NOTIMPL;
+        RETURN_HR_IF_NULL(E_POINTER, dpiX);
+        RETURN_HR_IF_NULL(E_POINTER, dpiY);
+        *dpiY = m_dpiY;
+        *dpiX = m_dpiX;
+        return S_OK;
     }
 
     HRESULT STDMETHODCALLTYPE CopyPalette(_In_ IWICPalette* palette) {
         UNIMPLEMENTED();
         return E_NOTIMPL;
     }
 
-    HRESULT STDMETHODCALLTYPE CopyPixels(_In_opt_ const WICRect* copyRect,
-                                         _In_ UINT stride,
-                                         _In_ UINT bufferSize,
-                                         _Out_writes_all_(cbBufferSize) BYTE* buffer) {
-        UNIMPLEMENTED();
-        return E_NOTIMPL;
-    }
-
     HRESULT STDMETHODCALLTYPE DisplayTexture(_Out_ IDisplayTexture** displayTexture) {
         RETURN_HR_IF_NULL(E_POINTER, displayTexture);
         *displayTexture = m_texture;
         return S_OK;
     }
 
 private:
+    void Init(_In_ WICPixelFormatGUID pixelFormat, _In_ UINT height, _In_ UINT width) {
+        m_dpiX = 96;
+        m_dpiY = 96;
+        m_pixelFormat = pixelFormat;
+        m_height = height;
+        m_width = width;
+        m_freeData = false;
+    }
+
     WICPixelFormatGUID m_pixelFormat;
     IDisplayTexture* m_texture;
```

Comment:
hmm technically this can be wrapped in unique_ptr for now.

---
In reply to: [88114838](https://github.com/Microsoft/WinObjC/pull/1371#discussion_r88114838) [](ancestors = 88114838,88100057)